doc/util.md

```diff
@@ -138,13 +148,78 @@ class MyBaseGame extends BaseGame {
 
 ## Extensions
 
-Flame bundles a collection of utility extensions, these extensions are meant to help the developer with shortcuts and converion methods, here you can find the summary of those extensions
+Flame bundles a collection of utility extensions, these extensions are meant to help the developer
+with shortcuts and conversion methods, here you can find the summary of those extensions.
 
-They can all be imported on `package:flame/extensions/...`
+They can all be imported from `package:flame/extensions.dart`
 
 ### Canvas
 
 Methods:
  - `scaleVector`: Just like `canvas scale` method, but takes a `Vector2` as an argument.
  - `translateVector`: Just like `canvas translate` method, but takes a `Vector2` as an argument.
- - `renderAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these functions to easily manipulate coordinates to render things on the correct places. They change the `Canvas` transformation matrix but reset afterwards.
+ - `renderAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these
+  functions to easily manipulate coordinates to render things on the correct places. They change the
+  `Canvas` transformation matrix but reset afterwards.
+
+### Color
+
+Methods:
+ - `darken`: Darken the shade of the color by an amount between 0 to 1.
+ - `brighten`: Brighten the shade of the color by an amount between 0 to 1.
+
+### Image
+
+Methods:
+ - `pixelsInUint8`: Retrieves the pixel data as a `Uint8List`, in the `ImageByteFormat.rawRgba`
+ pixel format, for the image.
+ - `getBoundingRect`: Get the bounding rectangle of the `Image` as a `Rect`.
+ - `size`: The size of an `Image` as `Vector2`.
+ - `darken`: Darken each pixel of the `Image` by an amount between 0 to 1.
+ - `brighten`: Brighten each pixel of the `Image` by an amount between 0 to 1.
+
+### Offset
+
+Methods;
+ - `toVector2`; Creates an `Vector2` from the `Offset`.
+ - `toSize`: Creates a `Size` from the `Offset`.
+ - `toPoint`: Creates a `Point` from the `Offset`.
+ - `toRect`: Creates a `Rect` starting in (0,0) and its bottom right corner is the [Offset].
+
+### Rect
+
+Methods:
+ - `toOffset`: Creates an `Offset` from the `Rect`.
+ - `toVector2`: Creates a `Vector2` starting in (0,0) and goes to the size of the `Rect`.
+ - `containsPoint` Whether this `Rect` contains a `Vector2` point or not.
+ - `intersectsSegment`; Whether the segment formed by two `Vector2`s intersects this `Rect`.
+ - `intersectsLineSegment`: Whether the `LineSegmet` intersects the `Rect`.
+ - `toVertices`: Turns the four corners of the `Rect` into a list of `Vector2`,
+
+### Size
+
+Methods:
+ - `toVector2`; Creates an `Vector2` from the `Size`.
+ - `toOffset`: Creates a `Offset` from the `Size`.
+ - `toPoint`: Creates a `Point` from the `Size`.
+ - `toRect`: Creates a `Rect` starting in (0,0) with the size of `Size`.
+
+### Vector2
```

Comment:
should probably mention here that this is a class that comes from the xxx package and we just add some extensions on top